### PR TITLE
fix(1749): Add ability to download artifact directory

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Should be able to download artifacts by directory.

## Objective

This PR adds ability to download artifact directory.

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/1749

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/artifact-tree/component.js [hunk 1]
```diff
@@ -88,6 +88,10 @@ export default Component.extend({
         const artifactPath = instance.get_path(node, '/');
 
         if (type === 'directory') {
+          this.setProperties({
+            href,
+            type: 'directory'
+          });
```
Comment (minghay) [inline app/components/artifact-tree/component.js:94]:
```suggestion
```

File: app/components/artifact-tree/component.js [hunk 2]
```diff
@@ -101,7 +105,8 @@ export default Component.extend({
           this.setProperties({
             href,
             iframeUrl: `${href}?type=preview`,
-            isModalOpen: true
+            isModalOpen: true,
+            type: 'file'
```
Comment (minghay) [inline app/components/artifact-tree/component.js:109]:
```suggestion
```

File: app/components/artifact-preview/component.js [hunk 3]
```diff
@@ -41,11 +41,17 @@ export default Component.extend({
 
   actions: {
     download() {
-      const downloadLink = this.iframeUrl.replace(
+      let downloadLink = this.iframeUrl.replace(
```
Comment (minghay) [inline app/components/artifact-preview/component.js:44]: Don't use the `iframeUrl` as it doesn't get set properly (I'm not currently sure why the tree structure isn't setting it properly for directories).

Inject the Ember router service into this component and you can use the `.currentRoute` property to extract the artifact path within it:
```javascript
  this.router.currentRoute.params.file_path
```

The artifact that is returned as a string will have trailing `/` for directories, so you can use that to do the rest of the needed work of recreating the proper URL.

One last thing, you'll need to recreate the base API URL like what is down in the `downloadAll` function just below this function

**Review by minghay — CHANGES_REQUESTED** [review]

File: app/components/artifact-preview/component.js (hunk 3 above)
Comment (tkyi) [reply app/components/artifact-preview/component.js:45]: updated

File: app/components/artifact-preview/component.js [hunk 4]
```diff
@@ -41,11 +42,20 @@ export default Component.extend({
...
         'type=preview',
         'type=download'
       );
 
+      if (this.type === 'directory') {
```
Comment (minghay) [inline app/components/artifact-preview/component.js:50]: `this.type` is always undefined.  The type was set in code that was used in a previous library.

**Review by minghay — APPROVED** [review]

**Pull request merged**